### PR TITLE
Add exercise recyclerview and viewmodel for exercises

### PR DESCRIPTION
**Pull request by brian-norman**
Adds a list of exercises (just Strings for now) and the ability to add new ones!

This screen will require a lot more work, but will break it up into multiple PRs for ease of review 😃

### CODE REVIEW

File: app/src/main/java/io/mochahub/powermeter/exercises/ExerciseAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package io.mochahub.powermeter.exercises
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/exercises/ExerciseAdapter.kt:1]: Moved exercise screen related code to an `exercises` package which I think we should do for each screen

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="Theme.MaterialComponents.DayNight">
```
Comment (brian-norman) [inline app/src/main/res/values/styles.xml:4]: This is really cool, turns out we get free night mode support here 😮 Just turn your device to night mode to see it in action!

File: app/src/main/java/io/mochahub/powermeter/exercises/ExerciseFragment.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_exercise, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```
Comment (zahin-mohammad) [inline app/src/main/java/io/mochahub/powermeter/exercises/ExerciseFragment.kt:27]: Is this preferred over doing the logic in `onActivityCreated`?

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]
Can you change the base branch to `brian/PM-13`. We should merge the incremental updates for the story to the story branch, then merge to master once done. Everything else  LGTM.

**Comment by zahin-mohammad** [comment]
Also, the Power Meter text in the app bar should remain white. I think this is a side effect from night mode.

**Comment by brian-norman** [comment]
> Can you change the base branch to `brian/PM-13`. We should merge the incremental updates for the story to the story branch, then merge to master once done. Everything else LGTM.

Why? I think it would be better if we don't do giant branch merges to master in the end, seems safer to me to keep incrementally updating master so we increase exposure of any side effects of our code. Also merge conflicts will be insane if we do 3 bulk merges 🤣

File: app/src/main/java/io/mochahub/powermeter/exercises/ExerciseFragment.kt (hunk 3 above)
Comment (brian-norman) [reply app/src/main/java/io/mochahub/powermeter/exercises/ExerciseFragment.kt:27]: Lol idk I think there is no difference 🤷‍♂

Comment (zahin-mohammad) [reply app/src/main/java/io/mochahub/powermeter/exercises/ExerciseFragment.kt:27]: kk. I'll use oneViewCreated on mine then, to keep it consistent.

**Comment by atulb98** [comment]
> > Can you change the base branch to `brian/PM-13`. We should merge the incremental updates for the story to the story branch, then merge to master once done. Everything else LGTM.
> 
> Why? I think it would be better if we don't do giant branch merges to master in the end, seems safer to me to keep incrementally updating master so we increase exposure of any side effects of our code. Also merge conflicts will be insane if we do 3 bulk merges

Okay, that makes sense. For the purpose of JIRA tracking all branches and PRs, can we follow the following convention:

For PRs, make the title: `[PM-13] Add exercise recyclerview and viewmodel for exercises`
For branches, make the name: `brian/PM-13-exercise-list`.

So, JIRA will keep track of all your PRs and branches for this story.

[PM-13]: https://mochahub.atlassian.net/browse/PM-13

**Pull request closed without merging**